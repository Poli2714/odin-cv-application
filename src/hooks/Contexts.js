import { createContext } from 'react';

const IndexContext = createContext({
  activeIndex: null,
  setActiveIndex: () => {},
});

const KeyContext = createContext({
  activeKey: {
    eduActiveKey: null,
    expActiveKey: null,
  },
  setActiveKey: () => {},
});

const EduErrMsgContext = createContext({
  showEduErrMessage: false,
  setShowEduErrMessage: () => {},
});

const ExpErrMsgContext = createContext({
  showExpErrMessage: false,
  setShowExpErrMessage: () => {},
});

const PersonalInfoContext = createContext({
  personalInfo: {
    profilePicture: '',
    fullName: '',
    currentJobTitle: '',
    email: '',
    mobile: '',
    city: '',
  },
  setPersonalInfo: () => {},
});

const SummaryContext = createContext({
  summary: '',
  setSummary: () => {},
});

const ExperienceContext = createContext({
  experiences: {
    id: {
      jobTitle: '',
      employer: '',
      jobStartDate: '',
      jobEndDate: '',
      jobDescription: '',
    },
  },
  setExperiences: () => {},
});

const EducationContext = createContext({
  educations: {
    id: {
      degree: '',
      university: '',
      eduStartDate: '',
      eduEndDate: '',
      eduInfo: '',
    },
  },
  setEducations: () => {},
});

export {
  EducationContext,
  EduErrMsgContext,
  ExperienceContext,
  ExpErrMsgContext,
  IndexContext,
  KeyContext,
  PersonalInfoContext,
  SummaryContext,
};
