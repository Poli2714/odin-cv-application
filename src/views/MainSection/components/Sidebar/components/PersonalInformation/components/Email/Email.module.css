.emailInput:not(:focus, :placeholder-shown) + .invalid {
  position: relative;

  &::after {
    color: var(--clr-warning);
    content: '⚠️ Not a valid email address';
    font-size: 0.85rem;

    position: absolute;
    bottom: -1.25rem;
    left: 0;
  }
}
