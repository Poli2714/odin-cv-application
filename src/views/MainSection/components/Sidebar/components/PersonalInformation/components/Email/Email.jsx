import { useContext } from 'react';
import isEmail from 'validator/lib/isEmail';

import { InputWrapper } from 'src/components/layout';
import { Label, TextInput } from 'src/components/forms';
import { PersonalInfoContext } from 'src/hooks/Contexts';
import styles from './Email.module.css';

function Email() {
  const { personalInfo, setPersonalInfo } = useContext(PersonalInfoContext);
  const { email } = personalInfo;

  const handleChange = e => {
    setPersonalInfo({ ...personalInfo, email: e.target.value });
  };

  return (
    <InputWrapper>
      <Label labelName="Email address" labelFor="email" />
      <TextInput
        id="email"
        clName={styles.emailInput}
        onChange={handleChange}
        pattern="^[a-zA-Z]+((-_\.)[a-zA-Z0-9]+)*[a-zA-Z0-9]*@[a-zA-Z0-9]+(-[a-zA-Z0-9]+)*[a-zA-Z0-9]+\.[a-zA-Z]{2,}(\.[a-zA-Z]{2,3})?$"
        placeholder="test@example.com"
        value={email}
      />
      {email.length > 0 && !isEmail(email) && (
        <div data-testid="warning" className={styles.invalid}></div>
      )}
    </InputWrapper>
  );
}

export default Email;
