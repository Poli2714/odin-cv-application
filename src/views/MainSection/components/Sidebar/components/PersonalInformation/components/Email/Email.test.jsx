import { expect, test, vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { useState } from 'react';

import Email from './Email';
import { PersonalInfoContext } from 'src/hooks/Contexts';

vi.mock('src/components/layout', () => ({
  InputWrapper: vi.fn(({ children }) => (
    <div data-testid="input-wrapper">{children}</div>
  )),
}));

vi.mock('src/components/forms', () => ({
  Label: vi.fn(() => <label htmlFor="email">Test label</label>),
}));

function MockParentComponent() {
  const [personalInfo, setPersonalInfo] = useState({ email: '' });

  return (
    <PersonalInfoContext.Provider value={{ personalInfo, setPersonalInfo }}>
      <Email />
    </PersonalInfoContext.Provider>
  );
}

test('renders Email', () => {
  render(<Email />);

  expect(screen.getByLabelText('Test label')).toMatchSnapshot();
});

test('updates email value as a user types', async () => {
  const user = userEvent.setup();

  render(<MockParentComponent />);
  const emailInput = screen.getByLabelText('Test label');
  expect(emailInput).toHaveValue('');

  await user.type(emailInput, 'elgun@test.com');
  expect(emailInput).toHaveValue('elgun@test.com');
  await user.clear(emailInput);
  expect(emailInput).toHaveValue('');
});
