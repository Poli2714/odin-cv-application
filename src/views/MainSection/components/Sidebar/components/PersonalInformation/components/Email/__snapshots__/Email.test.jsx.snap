// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders Email 1`] = `
<input
  id="email"
  type="text"
  value=""
/>
`;
